File: emulinker/src/main/java/org/emulinker/kaillera/master/MasterListStatsCollector.java
package org.emulinker.kaillera.master;

import java.util.*;
import javax.inject.Inject;
import javax.inject.Singleton;
import org.emulinker.kaillera.model.*;

@Singleton
public final class MasterListStatsCollector implements StatsCollector {
  private ArrayList<String> startedGamesList = new ArrayList<String>();

  @Inject
  MasterListStatsCollector() {}

  @Override
  public synchronized void gameStarted(KailleraServer server, KailleraGame game) {
    startedGamesList.add(game.getRomName());
  }

  @Override
  public synchronized List<String> getStartedGamesList() {
    return startedGamesList;
  }

  @Override
  public synchronized void clearStartedGamesList() {
    startedGamesList.clear();
  }
}

File: emulinker/src/main/java/org/emulinker/kaillera/master/MasterListStatsCollector.kt
package org.emulinker.kaillera.master

import javax.inject.Inject
import javax.inject.Singleton
import org.emulinker.kaillera.model.KailleraGame
import org.emulinker.kaillera.model.KailleraServer

@Singleton
class MasterListStatsCollector @Inject internal constructor() : StatsCollector {
  private val startedGamesList = mutableListOf<String>()

  @Synchronized
  override fun markGameAsStarted(server: KailleraServer, game: KailleraGame) {
    startedGamesList.add(game.romName!!)
  }

  @Synchronized
  override fun getStartedGamesList(): MutableList<String> {
    return startedGamesList
  }

  @Synchronized
  override fun clearStartedGamesList() {
    startedGamesList.clear()
  }
}
